File: src/main/java/fun/bookish/vertx/activemq/client/consumer/ActiveMQConsumer.java
package fun.bookish.vertx.activemq.client.consumer;


import fun.bookish.vertx.activemq.client.producer.ActiveMQProducer;
import fun.bookish.vertx.activemq.client.subscriber.ActiveMQSubscriber;
import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;
import io.vertx.core.json.JsonObject;

public interface ActiveMQConsumer {

    String getKey();

    void listen(Handler<AsyncResult<JsonObject>> messageHandler);

    void close();

    void close(Handler<AsyncResult<Void>> handler);

}

File: src/main/kotlin/cn/booklish/vertx/activemq/client/consumer/ActiveMQConsumer.kt
package cn.booklish.vertx.activemq.client.consumer

import io.vertx.core.AsyncResult
import io.vertx.core.Handler
import io.vertx.core.json.JsonObject


interface ActiveMQConsumer{

    val key:String

    fun listen(messageHandler: Handler<AsyncResult<JsonObject>>)

    fun close()

    fun close(handler: Handler<AsyncResult<Void>>)

}